### PR TITLE
ota-proxy cleanup

### PR DESCRIPTION
**Pull request by Bodong-Yang**
# Objective
1. stabilize and freeze ota-proxy's features. 
2. new way of ota-cache initialization.
3. documentation/comments in source code.
4. finish up type hinting.
5. other small refinements and cleanup.

# Changes
1. ota-cache now only being initialized in `ota_client_stub.py`, and being launched in `server_app.py` now.
2. explicitly specific which backend(libs) uvicorn uses.
3. fix `db.rotate_cache`: handling `None` result if query failed.

# Tickets
https://tier4.atlassian.net/browse/T4PB-17504

# TODO
- [x] fix `db.rotate_cache`
- [x] new way of ota-cache initialization
- [x] features stabilize
- [x] documentation in source code
- [x] simple documentation (README.md)
- [x] type hinting
- [x] regression test on local
- [x] regression test on VM

### CODE REVIEW

File: ota_proxy/README.md
Comment (obi-t4) [inline ota_proxy/README.md:3]: > Internal ECUs are not allowed to directly connect to the Internet

It depends on the user's environment. so,

`In case of the internal ECUs are not allowed to directly connect to the Internet, ota-proxy is introduced to allow internal ECUs to apply ota update.` should be fine.

**Review by obi-t4 — COMMENTED** [review]
Review is not finished yet but I left one comment.

**Review by obi-t4 — COMMENTED** [review]
I finished the review, it seems OK.

**Comment by Bodong-Yang** [comment]
> I finished the review, it seems OK.

Thank you! I will merge it after #131 is ready and merged.

File: ota_proxy/README.md
Comment (obi-t4) [reply ota_proxy/README.md:3]: @Bodong-Yang 
Can you fix this sentence?

Comment (Bodong-Yang) [reply ota_proxy/README.md:3]: OK!

**Comment by Bodong-Yang** [comment]
After regression test passed on VM, It is ready for merging.

**Comment by Bodong-Yang** [comment]
@obi-t4 regression test on VM is completed with the latest commit!

**Review by obi-t4 — APPROVED** [review]
:+1:

**Pull request merged**
